src/gluonts/itertools.py

```diff
@@ -289,6 +289,43 @@ def __iter__(self):
         return filter(self.fn, self.iterable)
 
 
+@dataclass
+class MixIterators:
+    iterators: List[Iterator]
+    probabilities: Optional[List[float]] = None
+    random_state: np.random.RandomState = np.random.RandomState()
+
+    def __post_init__(self):
+        if self.probabilities:
+            # copy the probabilities as we may modify it
+            self.probabilities = [prob for prob in self.probabilities]
+        else:
+            self.probabilities = [1.0 / len(self.iterators)] * len(
+                self.iterators
+            )
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        success = False
+        while not success:
+            probs = [
+                prob / sum(self.probabilities) for prob in self.probabilities
+            ]
+            idx = self.random_state.choice(range(len(self.iterators)), p=probs)
+
+            try:
+                data = next(self.iterators[idx])
+                success = True
+            except StopIteration:
+                self.probabilities[idx] = 0.0
```

Comment:
I don't think this will work, as the probabilities have to sum up to 1.

Reply:
It normalizes the probabilities in the beginning of the loop.